### PR TITLE
Configure Liveness probe

### PR DESCRIPTION
**Pull request by johnwatson484**
Add liveness probe to Mine Support service to allow cluster to restart on failure.

Unit tests converted from Lab to Jest.

https://eaflood.atlassian.net/browse/FPD-449

### CODE REVIEW

File: helm/values.yaml [hunk 1]
```diff
@@ -18,11 +18,19 @@ container:
...
+  
 ingress:
   serviceName: mine-support
   endPoint: mine-support
   server: defradev.com
 replicas: 1
 minReadySeconds: 5
 cookiePassword: passwordpasswordpasswordpasswordpassword
+
+livenessProbe:
```
Comment (markysoft) [inline helm/values.yaml]: there's a lot of whitespace after the `livenessProbe:` on this line

File: package.json [hunk 2]
```diff
@@ -6,16 +6,15 @@
   "scripts": {
     "build": "npm run build:css",
     "build:css": "node-sass --output-style=expanded --output=server/public/build/stylesheets client/sass",
-    "pretest": "npm run test:lint",
-    "start:watch": "nodemon --inspect=0.0.0.0 --ext js,njk --legacy-watch index.js",
-    "test": "npm run test:unit",
+    "test": "npm run test:lint && npm run test:unit",
```
Comment (markysoft) [inline package.json]: why has test reverted to using `&&` to run two commands, from the previous `pre-test` approach?

File: package.json [hunk 3]
```diff
@@ -6,16 +6,15 @@
...
     "test:lint": "standard",
-    "test:unit": "lab -c -r console -o stdout -r junit -o ./test-output/junit.xml"
+    "test:unit": "jest --forceExit",
+    "unit:watch": "jest --coverage=false --onlyChanged --watch --forceExit",
```
Comment (markysoft) [inline package.json]: are the forceExits required in the tests? The test script seem to run OK without it.

File: helm/values.yaml [hunk 4]
```diff
@@ -18,11 +18,19 @@ container:
...
 ingress:
   serviceName: mine-support
   endPoint: mine-support
   server: defradev.com
 replicas: 1
 minReadySeconds: 5
 cookiePassword: passwordpasswordpasswordpasswordpassword
+
+livenessProbe:          
+  path: /health
```
Comment (markysoft) [inline helm/values.yaml]: the convention in kubernetes is to call the end point `healthz`

File: package.json (hunk 2 above)
Comment (johnwatson484) [reply package.json]: To be consistent with the other services.  The test image doesn't run pretest so linting test wouldn't be run.

File: package.json (hunk 3 above)
Comment (johnwatson484) [reply package.json]: Will remove

**Comment by djbingham** [comment]
> Should information be added to the README decsribing the liveness probe?

Given that liveness and readiness probes are standard features in Kubernetes, I think documenting them would add unnecessary noise to the Readme.

File: helm/values.yaml (hunk 4 above)
Comment (johnwatson484) [reply helm/values.yaml]: I found a few different examples of what to call the end point.  healtz is in the official k8s site so can change it to that.

**Comment by johnwatson484** [comment]
> > Should information be added to the README decsribing the liveness probe?
> 
> Given that liveness and readiness probes are standard features in Kubernetes, I think documenting them would add unnecessary noise to the Readme.

Could just add a line stating what the endpoints are, rather than documenting them in detail?

File: package.json (hunk 2 above)
Comment (markysoft) [reply package.json]: Just asking the question, really. We don't document locking down the file system as readonly and other kubernetes best practices, so this wouldn't need explicitly mentioning either.

**Comment by markysoft** [comment]
Just asking the question, really. We don't document locking down the file system as readonly and other kubernetes best practices, so this wouldn't need explicitly mentioning either.

File: helm/values.yaml (hunk 1 above)
Comment (johnwatson484) [reply helm/values.yaml]: Fixed

**Comment by johnwatson484** [comment]
> Just asking the question, really. We don't document locking down the file system as readonly and other kubernetes best practices, so this wouldn't need explicitly mentioning either.

It's a fair question.  Probes aren't a requirement so I think it's worth highlighting that this service has one.  I've added a line to the Readme.

File: test/routes/healthz.test.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  beforeAll(async () => {
+    createServer = require('../../server')
+  })
+
+  beforeEach(async () => {
+    server = await createServer()
+    await server.initialize()
+  })
+
+  test('GET /liveness route returns 200', async () => {
```
Comment (Matthew-Collins-Defra) [inline test/routes/healthz.test.js]: The route is /healthz not /liveness, not functionally incorrect, but a bit misleading.

Comment (johnwatson484) [reply test/routes/healthz.test.js]: Good spot.  Fixed.

**Pull request merged**